viewing_party/party.py

```diff
@@ -1,23 +1,241 @@
+# to avoid naming confusing
+# "lod" in lod_<var.name> means list of dict
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''If those three attributes are truthy, then return a dictionary. This dictionary should...
+    Have three key-value pairs, with specific keys
+    The three keys should be "title", "genre", and "rating"
+    The values of these key-value pairs should be appropriate values
+    If title is falsy, genre is falsy, or rating is falsy, this function should return None'''
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    # create dict
+    movie_dict = {}
 
+    # falsy case
+    # even if one is falsy, the whole cond will return NONE
+    if not title or not genre or not rating:
+        return None
+    
+    movie_dict["title"] = title
+    movie_dict["genre"] = genre
+    movie_dict["rating"] = rating
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    '''
+    return user_data after adding movie to watched
+    An empty list represents that the user has no movies in their watched list
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watched = user_data["watched"]
+    lod_watched.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    return user_data after adding movie to watchlist
+    An empty list represents that the user has no movies in their watchlist
+    user_data = {
+        'watchlist': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+    lod_watchlist = user_data["watchlist"]
+    lod_watchlist.append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    - if title is in watchlist: remove from watchlist, add to watched, return user data
+    - if movie is not in user watchlist, return user data
+    user_data = {
+        'watched': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}],
+        'watchlist': [{"title": "", "genre": "", "rating": <float>}, {...}, {...}, {...}, {...}, {...}]
+        }
+    '''
+
+    lod_watchlist = user_data["watchlist"]
+    lod_watched = user_data["watched"]
+    # returns [ {movie_dict} , {movie_dict} , ...]

```

Comment:
This comment could be moved into the function's docstring and then you can remove it from the function body